spacify-ui/src/components/rooms/Rooms.jsx

```diff
@@ -1,36 +1,129 @@
 import React, {useEffect, useState} from "react";
 import "./rooms.css";
+import {ALL_SUBSCRIBED_ROOMS_API, USER_NAME_KEY, UPDATE_SUBSCRIBED_ROOMS_API} from "../../endpoints";
 
 const Rooms = () => {
   const [rooms, setRooms] = useState([]);
+  const [selectedRoom, setSelectedRoom] = useState(null); // Selected room for the pop-up form
+  const [phoneNo, setPhoneNo] = useState("");               // Phone number for WhatsApp subscription
+  const [showPhoneNumberInput, setShowPhoneNumberInput] = useState(false); // Control visibility of phone number input
+  const [roomId, setRoomID] = useState("");
+  const savedRoomId = null;
+  const savedPhoneNo = null;
+  const [response, setResponse] = useState({header: "", show: false,  responseMessage: ""});
+  const axios = require('axios');
+
 
-  // Use this useEffect hook to fetch the list of rooms from the API endpoint
-  // useEffect(() => {
-  //   axios.get("/api/my-rooms")
-  //     .then(res => setRooms(res.data))
-  //     .catch(err => console.log(err));
-  // }, []);
 
-  // For now, we'll use a mocked list of rooms
   useEffect(() => {
-    setRooms([
-      { id: 1, name: "VP-C-120", type: "study" },
-      { id: 2, name: "DBH-1200", type: "office" },
-      { id: 3, name: "VP-D-130", type: "study" }
-    ]);
+
+//     let userId = global.sessionStorage.getItem(USER_NAME_KEY);
+    let url = ALL_SUBSCRIBED_ROOMS_API + "kshatris";        //HARDCODED change later
+    console.log(url);
+     fetch(url)
+              .then((res) => res.json())
+              .then((data) => {
+              console.log(data);
+                    setRooms(data);
+                    });
+
+
+//     setRooms([
+//       { id: 1, name: "VP-C-120", type: "study" },
+//       { id: 2, name: "DBH-1200", type: "office" },
+//       { id: 3, name: "VP-D-130", type: "study" }
+//     ]);
   }, []);
 
+  const handleRoomSelect = (room) => {
+    setSelectedRoom(room);
+  };
+
+  const handleCloseForm = () => {
+    setSelectedRoom(null);
+    setShowPhoneNumberInput(false);
+    setPhoneNo("");
+  };
+
+  const handleSubscribe = () => {
+    setShowPhoneNumberInput(true);
+  };
+
+  const handlePhoneNoChange = (e) => {
+    setPhoneNo(e.target.value);
+  };
+
+  const handleConfirmSubscription = () => {
+    console.log(`Subscribing to WhatsApp for room: ${selectedRoom.roomDescription}`);
+    console.log(`Phone number: ${phoneNo}`);
+    setShowPhoneNumberInput(false);
+
+    console.log("Saved phone number:", savedPhoneNo);
+    console.log("Saved room ID:", savedRoomId);
```

Comment:
we need to check if the whatsapp number is required for the user and only then take the number